### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by osk2**
Hi, I just made some improvements 😀 

- According to [AMP tutorial](https://www.ampproject.org/docs/get_started/create/basic_markup), `charset` should be the first child of `head` and JS lib should be the second.
I adjusted it to match requirement
- And remove invalid `img` tag.
In my case, some auto-generated HTML will contain invalid tag like ``
- Complete `usage` in readme
- Update tests

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,10 +35,14 @@ module.exports = function(html, options) {
 
   /* head */
 
+  /* main amp library */
+  $('head script[src="https://cdn.ampproject.org/v0.js"]').remove();
+  $('head').prepend('<script async src="https://cdn.ampproject.org/v0.js"></script>');
```
Comment (rkazakov) [inline index.js]: Should `amp lib` go after `meta charset`?

**Comment by rkazakov** [comment]
Hi @osk2, thanks for the contribution! Please check the comment above.

File: index.js (hunk 1 above)
Comment (osk2) [inline index.js]: I use `prepend` so that `meta` will be the first element of `head`, and `script` will be the second

Comment (rkazakov) [inline index.js]: Cool! 👍

**Pull request merged**